src/executor.ts

```diff
@@ -109,13 +110,15 @@ export class IdempotentExecutor {
           // If the action has already been executed, replay the result.
           if (cachedResult) {
             if (cachedResult.type === 'error') {
-              this.replayCachedError(
-                idempotencyKey,
-                errorSerializer,
-                cachedResult.error,
-                options?.onErrorReplay,
-              );
-            }
+              if (!ignoreErrors) {
```

Comment:
This check needs to be made earlier in the process -- before the error is stored (see `cacheResult` method), rather than before it's replayed.